pkg/chrootarchive/archive_test.go

```diff
@@ -39,6 +41,80 @@ func UntarPath(src, dst string) error {
 func CopyWithTar(src, dst string) error {
 	return chrootArchiver.CopyWithTar(src, dst)
 }
+func TestUnTarUncompressed(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "docker-TestUntar")
+	require.NoError(t, err)
+
+	defer os.RemoveAll(tempDir)
+
+	destination := filepath.Join(tempDir, "src")
+
+	err = system.MkdirAll(destination, 0700)
+	require.NoError(t, err)
+
+	err = ioutil.WriteFile(filepath.Join(destination, "test"), []byte("test un compressed"), 0644)
+	require.NoError(t, err)
+
+	err = UntarUncompressed(nil, destination, &archive.TarOptions{ExcludePatterns: []string{"testuncompresss"}})
```

Comment:
Although it doesn't really _hurt_, all the exclusion patterns seem to be covered already by other tests (the only different between `UntarUncompressed` and `Untar` is this bit https://github.com/moby/moby/blob/93fbdb69acf9248283a91a1c5c6ea24711c26eda/pkg/chrootarchive/archive.go#L60-L67, of which only the error-case seems to be not covered.